src/sidebar/services/test/threads-test.js

```diff
@@ -0,0 +1,88 @@
+import threadsService from '../threads';
+
+const NESTED_THREADS = {
+  id: 'top',
+  children: [
+    {
+      id: '1',
+      children: [
+        { id: '1a', children: [{ id: '1ai', children: [] }] },
+        { id: '1b', children: [] },
+        {
+          id: '1c',
+          children: [{ id: '1ci', children: [] }],
+        },
+      ],
+    },
+    {
+      id: '2',
+      children: [
+        { id: '2a', children: [] },
+        {
+          id: '2b',
+          children: [
+            { id: '2bi', children: [] },
+            { id: '2bii', children: [] },
+          ],
+        },
+      ],
+    },
+    {
+      id: '3',
+      children: [],
+    },
+  ],
+};
+
+describe('threads', function() {
+  let fakeStore;
+  let service;
+
+  beforeEach(() => {
+    fakeStore = {
+      setForceVisible: sinon.stub(),
+    };
+    service = threadsService(fakeStore);
+  });
+
+  describe('#forceVisible', () => {
```

Comment:
I'm the first to admit that these tests are a bit ham-fisted, but I didn't immediately conceive of a brilliant better way.